tests/unit-core/test-arraybuffer.c

```diff
@@ -153,15 +153,37 @@ static void test_write_with_offset (uint8_t offset) /**< offset for buffer write
 
 static bool callback_called = false;
 static bool detach_free_callback_called = false;
+static bool user_callback_called = false;
+static uint32_t external_buffer_size = 15;
 
-static void test_free_cb (void *buffer) /**< buffer to free (if needed) */
+static void
+test_free_cb (uint32_t length, /**< length of buffer */
+              void *buffer, /**< buffer to free (if needed) */
+              void *user_p) /**< user pointer */
 {
+  JERRY_UNUSED (length);
+  JERRY_UNUSED (user_p);
```

Comment:
the length and user_p should be checked in some way. e.g. length should be 15 or its other values, and user_p could be compared to some global values.